ui/pages/confirmations/components/confirm/footer/footer.tsx

```diff
@@ -226,41 +229,56 @@ const Footer = () => {
     hardwareWalletRequiresConnection ||
     isGaslessLoading;
 
-  const onSubmit = useCallback(() => {
+  const onSubmit = useCallback(async () => {
     if (!currentConfirmation) {
       return;
     }
 
     if (isAddEthereumChain) {
-      onAddEthereumChain();
+      await onAddEthereumChain();
+      history.push(DEFAULT_ROUTE);
     } else if (isTransactionConfirmation) {
-      onTransactionConfirm();
+      await onTransactionConfirm();
+      navigateNext(currentConfirmation.id);
     } else {
-      dispatch(resolvePendingApproval(currentConfirmation.id, undefined));
+      await dispatch(resolvePendingApproval(currentConfirmation.id, undefined));
+      navigateNext(currentConfirmation.id);
```

Comment:
Prevents race condition where navigation happens before approval completes, causing spinner to hang in sidepanel.